spec/player_input_spec.rb

```diff
@@ -0,0 +1,35 @@
+require_relative "../lib/player_input"
+
+RSpec.describe PlayerInput do
```

Comment:
These tests still seem pretty integration focused. We need to inspect the output in order to test whether the coordinates are validated. Earlier I was talking about a possible `Turn` class and a wrapper class for input, but it might be that there is a data validation class hidden in there.

Reply:
Updated to inject a view object instead of the output stream in 5d2a0380db819c28395dc3e2766e56b18889b748. It does feel much less integration focused, but I don't really like the heavy use of mocking - I don't feel super confident the test has me covered. I'm gonna keep poking at it, but figured I'd share what's there for now.